topostats/filters.py

```diff
@@ -160,30 +160,30 @@ def flatten_image(self, image: np.ndarray, mask: np.ndarray = None) -> np.ndarra
         image = self.remove_tilt(image, mask)
         return image
 
-    def row_col_quantiles(self, image: np.array, mask: np.array = None) -> np.array:
-        """Returns the height value quantiles for the rows and columns.
+    def row_col_medians(self, image: np.array, mask: np.array = None) -> np.array:
+        """Returns the height value medians for the rows and columns.
 
         Returns
         -------
 
-        Returns two Numpy arrays, the first is the row height value quantiles, the second the column height value
-        quantiles.
+        Returns two Numpy arrays, the first is the row height value medians, the second the column height value
+        medians.
         """
         if mask is not None:
-            image = np.ma.masked_array(image, mask=mask, fill_value=np.nan)
+            image = np.ma.masked_array(image, mask=mask, fill_value=np.nan).filled()
             LOGGER.info(f"[{self.filename}] : Masking enabled")
         else:
             LOGGER.info(f"[{self.filename}] : Masking disabled")
-        quantiles = {}
-        quantiles["rows"] = np.quantile(image, [0.25, 0.5, 0.75], axis=1).T
-        quantiles["cols"] = np.quantile(image, [0.25, 0.5, 0.75], axis=0).T
-        LOGGER.info(f"[{self.filename}] : Row and column quantiles calculated.")
-        return quantiles
+        medians = {}
+        medians["rows"] = np.nanmedian(image, axis=1).T
```

Comment:
Minor but I don't think you need to transpose the array since its 1D here, sorry for not clocking this earlier.

```suggestion
        medians["rows"] = np.nanmedian(image, axis=1)
```

Same applies to next line.

Reply:
Done 👍